flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/SlotSharingSlotAllocator.java

```diff
@@ -301,27 +301,24 @@ public Collection<ExecutionVertexID> getContainedExecutionVertices() {
 
     private static class SlotSharingGroupMetaInfo {
 
-        private final int minLowerBound;
         private final int maxUpperBound;
-        private final int maxLowerUpperBoundRange;
+        private final int minResourceRequirement;
 
-        private SlotSharingGroupMetaInfo(
-                int minLowerBound, int maxUpperBound, int maxLowerUpperBoundRange) {
-            this.minLowerBound = minLowerBound;
+        private SlotSharingGroupMetaInfo(int maxUpperBound, int minResourceRequirement) {
             this.maxUpperBound = maxUpperBound;
-            this.maxLowerUpperBoundRange = maxLowerUpperBoundRange;
-        }
-
-        public int getMinLowerBound() {
-            return minLowerBound;
+            this.minResourceRequirement = minResourceRequirement;
         }
 
         public int getMaxUpperBound() {
             return maxUpperBound;
         }
 
-        public int getMaxLowerUpperBoundRange() {
-            return maxLowerUpperBoundRange;
+        public int getMaxUpperMinRequirementBoundRange() {
+            return maxUpperBound - minResourceRequirement;
+        }
+
+        public int getMinResourceRequirement() {
```

Comment:
I'd prefer if we'd name this `getMaxLowerBound` because that's exactly what it is. In this area of the code it is always good to use a consistent terminology and be as explicit as possible.